crates/lang/macro/src/lib.rs

```diff
@@ -632,5 +635,435 @@ pub fn test(attr: TokenStream, item: TokenStream) -> TokenStream {
+///     ///
+///     /// # Note
+///     ///
+///     /// Assumes to never fail the call and therefore always returns `Option<Access`.
+///     #[ink(extension = 4, expect_ok, expect_output)]
+///     fn access(key: &[u8]) -> Option<Access>;
+///
+///     /// Unlocks previously aquired permission to access key.
+///     ///
+///     /// # Errors
+///     ///
+///     /// If the permission was not granted.
+///     ///
+///     /// # Note
+///     ///
+///     /// Assumes the call to never fail and therefore does _NOT_ require `UnlockAccessError`
+///     /// to implement `From<Self::ErrorCode>` as in the `read_small` method above.
+///     #[ink(extension = 5, expect_output)]
+///     fn unlock_access(key: &[u8], access: Access) -> Result<(), UnlockAccessError>;
+/// }
+/// # #[derive(scale::Encode, scale::Decode, scale_info::TypeInfo)]
+/// # pub enum ReadWriteErrorCode {
+/// #     InvalidKey,
+/// #     CannotWriteToKey,
+/// #     CannotReadFromKey,
+/// # }
+/// # #[derive(scale::Encode, scale::Decode, scale_info::TypeInfo)]
+/// # pub enum ReadWriteError {
+/// #     ErrorCode(ReadWriteErrorCode),
+/// #     BufferTooSmall { required_bytes: u32 },
+/// # }
+/// # impl From<ReadWriteErrorCode> for ReadWriteError {
+/// #     fn from(error_code: ReadWriteErrorCode) -> Self {
+/// #         Self::ErrorCode(error_code)
+/// #     }
+/// # }
+/// # impl From<scale::Error> for ReadWriteError {
+/// #     fn from(_: scale::Error) -> Self {
+/// #         panic!("encountered unexpected invalid SCALE encoding")
+/// #     }
+/// # }
+/// # #[derive(scale::Encode, scale::Decode, scale_info::TypeInfo)]
+/// # pub struct UnlockAccessError {
+/// #     reason: String,
+/// # }
+/// # impl From<scale::Error> for UnlockAccessError {
+/// #     fn from(_: scale::Error) -> Self {
+/// #         panic!("encountered unexpected invalid SCALE encoding")
+/// #     }
+/// # }
+/// # #[derive(scale::Encode, scale::Decode, scale_info::TypeInfo)]
+/// # pub enum Access {
+/// #     ReadWrite,
+/// #     ReadOnly,
+/// #     WriteOnly,
+/// # }
+/// # impl ink_env::chain_extension::FromStatusCode for ReadWriteErrorCode {
+/// #     fn from_status_code(status_code: u32) -> Result<(), Self> {
+/// #         match status_code {
+/// #             0 => Ok(()),
+/// #             1 => Err(Self::InvalidKey),
+/// #             2 => Err(Self::CannotWriteToKey),
+/// #             3 => Err(Self::CannotReadFromKey),
+/// #             _ => panic!("encountered unknown status code"),
+/// #         }
+/// #     }
+/// # }
+/// ```
+///
+/// All the error types and other utility types used in the chain extension definition
+/// above are often required to implement various traits such as SCALE's `Encode` and `Decode`
+/// as well as `scale-info`'s `TypeInfo` trait.
+///
+/// A full example of the above chain extension definition can be seen
+/// [here](https://github.com/paritytech/ink/blob/robin-impl-chain-extension/crates/lang/macro/tests/ui/chain_extension/simple.rs).
```

Comment:
```suggestion
/// [here](https://github.com/paritytech/ink/blob/master/crates/lang/macro/tests/ui/chain_extension/simple.rs).
```

Reply:
Maybe revision is a more stable version to guarantee that we point to an actual state of the file that we know precisely.